### PR TITLE
update documentation for deprecated matplotlib references

### LINKED ISSUE #37603: Documentation References Deprecated URL for 3D Plotting in Matplotlib
**Describe the bug**
 The Matplotlib 3D plotting tutorial referenced in this documentation : https://docs.mantidproject.org/nightly/plotting/3DPlotsHelp.html is deprecated
 
**To Reproduce**
1. visit https://docs.mantidproject.org/nightly/plotting/3DPlotsHelp.html 
2. click "3D Plotting in Matplotlib"

**Expected behavior**
The link is expected to point to the new 3D Matplotlib plotting URL: https://matplotlib.org/stable/users/explain/toolkits/mplot3d.html.

### PR DESCRIPTION
**Pull request by yusufjimoh**
#### Summary of work

update deprecated matplotlib references in mantid documentation to the latest

Fixes #37603. 

### To test:

Build the `docs-html` target and check that the changes render correctly

*This does not require release notes* because **it is only a documentation update.**

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by rbauststfc — COMMENTED** [review]
Looks like a really thorough update of all the relevant URLs. I've built both the user and dev docs and the changes are rendering well with all the new links working.

I noticed a very small typo that is unrelated to these changes but might be good to fix while we're here, if that's OK.

The other thing that might just be good to discuss quickly with the Core team (if you haven't already) is that I think at the moment we're on Matplotlib 3.7. Despina is working on https://github.com/mantidproject/mantid/pull/38077 to move us to 3.8. The current stable version is 3.9. I've noticed that a lot of the URL updates in this PR are pointing us to the docs for the stable version. The benefit of not specifying the version in the links is obviously that they won't need updating every time we update Matplotlib, but there is a small risk that we're pointing to documentation that isn't totally accurate for the version we're on. Might be good to just get a second opinion from Core to confirm we're happy with this.

**Review by jhaigh0 — APPROVED** [review]
It's a good idea to use these stable links, we won't have to keep updating them and we try to keep as close to the stable release as possible.

**Pull request merged**
